pub-matic-htb.js

```diff
@@ -462,21 +462,23 @@ function PubMaticHtb(configs) {
             }
 
             for (var i = 0; i < bids.length; i++) {
-                sizes = curReturnParcel.xSlotRef.sizes[0];
-
-                /**
-                 * This section maps internal returnParcels and demand returned from the bid request.
-                 * In order to match them correctly, they must be matched via some criteria. This
-                 * is usually some sort of placements or inventory codes. Please replace the someCriteria
-                 * key to a key that represents the placement in the configuration and in the bid responses.
-                */
-                if (bids[i].impid === curReturnParcel.htSlot.getId()) {
-                    if (parseInt(bids[i].w, 10) === parseInt(sizes[0], 10)
-                        && parseInt(bids[i].h, 10) === parseInt(sizes[1], 10)) {
-                        curBid = bids[i];
-                        bids.splice(i, 1);
-
-                        break;
+                for (var index = 0; index < curReturnParcel.xSlotRef.sizes.length; index++) {
+                    sizes = curReturnParcel.xSlotRef.sizes[index];
+
+                    /**
+                     * This section maps internal returnParcels and demand returned from the bid request.
+                     * In order to match them correctly, they must be matched via some criteria. This
+                     * is usually some sort of placements or inventory codes. Please replace the someCriteria
+                     * key to a key that represents the placement in the configuration and in the bid responses.
+                    */
+                    if (bids[i].impid === curReturnParcel.htSlot.getId()) {
+                        if (parseInt(bids[i].w, 10) === parseInt(sizes[0], 10)
+                            && parseInt(bids[i].h, 10) === parseInt(sizes[1], 10)) {
+                            curBid = bids[i];
+                            bids.splice(i, 1);
```

Comment:
why do we use bids.splice(i, 1), what if we do not alter bids array?
if we do not alter bids array then we can avoid calculating bids length in loop

Reply:
this was of the code was provided by index in the template, and i have kept it as it is. I have just added the condition on which we match the bids.
Since the code loops over returnparcels and matches them with the bids received, we update the bids array, so that we dont check against the same bids for subsequent returnParcels.

Reply:
as every bids.impid should be unique can you check if we can avoid splicing?

Reply:
In this check if (bids[i].impid === curReturnParcel.htSlot.getId())
the ID returned by curReturnParcel.htSlot.getId() is same for all return parcels generated. Hence the splicing of bids is required. 
This happens because, in the test case config, they map both the following xslots to the same slot:
xSlots: {
            1: { adUnitName: '/43743431/DMDemo', sizes: [[728, 90], [160, 600], [300, 250]] },
            2: { adUnitName: '/43743431/DMDemo1',  sizes: [[728, 90], [800, 250], [300, 250] ] }
}
config.mapping = {
      htSlotDesktopA: ['1'  , '2']
};
This also means that in the translator request, we get 2 impression objects with same imp[i].id.
Although on production that is not the case. We are getting different imp.ids in each impression object.

Reply:
ok, then let's not invest more time on this comment.